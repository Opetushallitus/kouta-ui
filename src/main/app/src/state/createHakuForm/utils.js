import get from 'lodash/get';
import pick from 'lodash/pick';

import {
  getKoutaDateString,
  isNumeric,
  formatKoutaDateString,
} from '../../utils';

const DATE_FORMAT = 'DD.MM.YYYY HH:mm';

const getDateTimeValues = dateString => {
  if (!dateString) {
    return { date: '', time: '' };
  }

  const formattedDate = formatKoutaDateString(dateString, DATE_FORMAT);

  const [d = '', t = ''] = formattedDate.split(' ');

  return { date: d, time: t };
};

const getKoutaDateStringByDateTime = ({ date = '', time = '' }) => {
  const [day, month, year] = date.split('.');

  if (!isNumeric(day) || !isNumeric(month) || !isNumeric(year)) {
    return null;
  }

  const [hour, minute] = time.split(':');

  return getKoutaDateString({
    day,
    month,
    year,
    hour: hour || 0,
    minute: minute || 0,
  });
};

export const getHakuByValues = values => {
  const alkamiskausiKoodiUri = get(values, 'aikataulut.kausi') || null;
  const alkamisvuosi = parseInt(get(values, 'aikataulut.vuosi'));
  const kielivalinta = get(values, 'kieliversiot.languages') || [];

  const hakutapaKoodiUri = get(values, 'hakutapa.tapa') || null;

  const hakuajat = (get(values, 'aikataulut.hakuaika') || []).map(
    ({ fromDate, fromTime, toDate, toTime }) => ({
      alkaa: getKoutaDateStringByDateTime({
        date: fromDate,
        time: fromTime,
      }),
      paattyy: getKoutaDateStringByDateTime({
        date: toDate,
        time: toTime,
      }),
    }),
  );

  const hakulomaketyyppi = get(values, 'hakulomake.lomaketyyppi') || null;

  const hakulomake = get(values, 'hakulomake.lomake') || null;

  const hakukohteenLiittamisenTakaraja =
    get(values, 'aikataulut.liittäminen_pvm') &&
    get(values, 'aikataulut.liittäminen_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.liittäminen_pvm,
          time: values.aikataulut.liittäminen_aika,
        })
      : null;

  const ajastettuJulkaisu =
    get(values, 'aikataulut.julkaisu_pvm') &&
    get(values, 'aikataulut.julkaisu_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.julkaisu_pvm,
          time: values.aikataulut.julkaisu_aika,
        })
      : null;

  const nimi = pick(get(values, 'nimi.nimi') || null, kielivalinta);

  const kohdejoukkoKoodiUri = get(values, 'kohdejoukko.kohde') || null;

  const kohdejoukonTarkenneKoodiUri = null;

  const metadata = {
    tulevaisuudenAikataulu: (get(values, 'aikataulut.aikataulu') || []).map(
      ({ fromDate, fromTime, toDate, toTime }) => ({
        alkaa: getKoutaDateStringByDateTime({
          date: fromDate,
          time: fromTime,
        }),
        paattyy: getKoutaDateStringByDateTime({
          date: toDate,
          time: toTime,
        }),
      }),
    ),
    yhteystieto: {
      nimi: pick(get(values, 'yhteystiedot.nimi') || null, kielivalinta),
      titteli: pick(get(values, 'yhteystiedot.titteli') || null, kielivalinta),
      sahkoposti: pick(get(values, 'yhteystiedot.email') || null, kielivalinta),
      puhelinnumero: pick(
        get(values, 'yhteystiedot.puhelin') || null,
        kielivalinta,
      ),
    },
  };

  const hakukohteenMuokkaamisenTakaraja =
    get(values, 'aikataulut.muokkaus_pvm') &&
    get(values, 'aikataulut.muokkaus_aika')
      ? getKoutaDateStringByDateTime({
          date: values.aikataulut.muokkaus_pvm,
          time: values.aikataulut.muokkaus_aika,
        })
      : null;

  return {
    alkamiskausiKoodiUri,
    kielivalinta,
    hakutapaKoodiUri,
    hakuajat,
    hakukohteenLiittamisenTakaraja,
    nimi,
    kohdejoukkoKoodiUri,
    kohdejoukonTarkenneKoodiUri,
    hakulomaketyyppi,
    metadata,
    hakukohteenMuokkaamisenTakaraja,
    ajastettuJulkaisu,
    alkamisvuosi,
    hakulomake,
  };
};

export const getValuesByHaku = haku => {
  const {
    alkamiskausiKoodiUri = '',
    alkamisvuosi = '',
    hakuajat = [],
    hakutapaKoodiUri = '',
    kohdejoukkoKoodiUri = '',
    hakulomaketyyppi = '',
    hakukohteenLiittamisenTakaraja,
    hakukohteenMuokkaamisenTakaraja,
    ajastettuJulkaisu,
    kielivalinta = [],
    nimi = {},
    metadata = {},
  } = haku;

  const { yhteystieto = {} } = metadata;

  const { tulevaisuudenAikataulu = [] } = metadata;

  const { date: liittäminen_pvm, time: liittäminen_aika } = getDateTimeValues(
    hakukohteenLiittamisenTakaraja,
  );

  const { date: muokkaus_pvm, time: muokkaus_aika } = getDateTimeValues(
    hakukohteenMuokkaamisenTakaraja,
  );

  const { date: julkaisu_pvm, time: julkaisu_aika } = getDateTimeValues(
    ajastettuJulkaisu,
  );

  return {
    nimi: {
      nimi: nimi,
    },
    kieliversiot: {
      languages: kielivalinta,
    },
    aikataulut: {
      kausi: alkamiskausiKoodiUri,
      vuosi: alkamisvuosi ? alkamisvuosi.toString() : '',
      hakuaika: (hakuajat || []).map(({ alkaa, paattyy }) => {
        const { date: fromDate, time: fromTime } = getDateTimeValues(alkaa);
        const { date: toDate, time: toTime } = getDateTimeValues(paattyy);

        return {
          fromDate,
          fromTime,
          toDate,
          toTime,
        };
      }),
      aikataulu: (tulevaisuudenAikataulu || []).map(({ alkaa, paattyy }) => {
        const { date: fromDate, time: fromTime } = getDateTimeValues(alkaa);
        const { date: toDate, time: toTime } = getDateTimeValues(paattyy);

        return {
          fromDate,
          fromTime,
          toDate,
          toTime,
        };
      }),
      liittäminen_pvm: liittäminen_pvm,
      liittäminen_aika: liittäminen_aika,
      muokkaus_pvm: muokkaus_pvm,
      muokkaus_aika: muokkaus_aika,
      julkaisu_pvm: julkaisu_pvm,
      julkaisu_aika: julkaisu_aika,
    },
    hakutapa: {
      tapa: hakutapaKoodiUri,
    },
    kohdejoukko: {
      kohde: kohdejoukkoKoodiUri,
    },
    hakulomake: {
      lomaketyyppi: hakulomaketyyppi,
    },
    yhteystiedot: {
      nimi: get(yhteystieto, 'nimi') || {},
      titteli: get(yhteystieto, 'titteli') || {},
      email: get(yhteystieto, 'sahkoposti') || {},
      puhelin: get(yhteystieto, 'puhelinnumero') || {},
    },
  };
};
