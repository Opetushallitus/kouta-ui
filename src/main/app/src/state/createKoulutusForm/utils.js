import get from 'lodash/get';
import pick from 'lodash/pick';

import { JULKAISUTILA, KORKEAKOULUKOULUTUSTYYPIT } from '../../constants';
import { ErrorBuilder } from '../../validation';

const getKielivalinta = values => get(values, 'kieliversiot.languages') || [];

export const getKoulutusByValues = values => {
  const kielivalinta = getKielivalinta(values);
  const tarjoajat = get(values, 'tarjoajat') || [];
  const koulutusKoodiUri = get(values, 'information.koulutus.value') || null;
  const koulutustyyppi = get(values, 'type.type') || null;
  const osiot = get(values, 'lisatiedot.osiot') || [];
  const osioKuvaukset = get(values, 'lisatiedot.osioKuvaukset') || {};

  const osiotWithKuvaukset = osiot.map(({ value }) => ({
    otsikkoKoodiUri: value,
    teksti: pick(osioKuvaukset[value] || {}, kielivalinta),
  }));

  const kuvaus = pick(get(values, 'description.kuvaus') || {}, kielivalinta);
  const nimi = pick(get(values, 'information.nimi') || {}, kielivalinta);
  const opintojenLaajuusKoodiUri =
    get(values, 'information.opintojenLaajuus.value') || null;

  const tutkintonimikeKoodiUrit = (
    get(values, 'information.tutkintonimike') || []
  ).map(({ value }) => value);

  const koulutusalaKoodiUrit = (
    get(values, 'information.koulutusalat') || []
  ).map(({ value }) => value);

  const kuvauksenNimi = pick(
    get(values, 'description.nimi') || {},
    kielivalinta,
  );

  const julkinen = Boolean(get(values, 'nakyvyys.julkinen'));

  return {
    kielivalinta,
    tarjoajat,
    koulutusKoodiUri,
    koulutustyyppi,
    nimi,
    julkinen,
    metadata: {
      tyyppi: koulutustyyppi,
      lisatiedot: osiotWithKuvaukset,
      kuvaus,
      opintojenLaajuusKoodiUri,
      tutkintonimikeKoodiUrit,
      kuvauksenNimi,
      koulutusalaKoodiUrit,
    },
  };
};

export const getValuesByKoulutus = koulutus => {
  const {
    kielivalinta = [],
    koulutusKoodiUri = '',
    koulutustyyppi = '',
    tarjoajat = [],
    metadata = {},
    nimi = {},
    julkinen = false,
  } = koulutus;

  const {
    lisatiedot = [],
    kuvaus = {},
    opintojenLaajuusKoodiUri = '',
    tutkintonimikeKoodiUrit = [],
    kuvauksenNimi = {},
    koulutusalaKoodiUrit = [],
  } = metadata;

  const osiot = lisatiedot
    .filter(({ otsikkoKoodiUri }) => !!otsikkoKoodiUri)
    .map(({ otsikkoKoodiUri }) => ({ value: otsikkoKoodiUri }));

  const osioKuvaukset = lisatiedot.reduce((acc, curr) => {
    if (curr.otsikkoKoodiUri) {
      acc[curr.otsikkoKoodiUri] = curr.teksti || {};
    }

    return acc;
  }, {});

  return {
    kieliversiot: {
      languages: kielivalinta,
    },
    tarjoajat,
    information: {
      nimi,
      koulutus: {
        value: koulutusKoodiUri,
      },
      opintojenLaajuus: {
        value: opintojenLaajuusKoodiUri,
      },
      tutkintonimike: tutkintonimikeKoodiUrit.map(value => ({ value })),
      koulutusalat: koulutusalaKoodiUrit.map(value => ({ value })),
    },
    type: {
      type: koulutustyyppi,
    },
    lisatiedot: {
      osioKuvaukset,
      osiot,
    },
    description: {
      kuvaus,
      nimi: kuvauksenNimi,
    },
    nakyvyys: {
      julkinen,
    },
  };
};

const validateEssentials = ({ errorBuilder }) => {
  return errorBuilder
    .validateArrayMinLength('kieliversiot.languages', 1)
    .validateExistence('information.koulutus');
};

const validateCommon = ({ errorBuilder }) => {
  return errorBuilder.validateArrayMinLength('tarjoajat', 1);
};

const validateKorkeakoulu = ({ values, errorBuilder }) => {
  const kielivalinta = getKielivalinta(values);

  return errorBuilder.validateTranslations('description.kuvaus', kielivalinta);
};

export const validate = ({ values, koulutustyyppi, tila }) => {
  let errorBuilder = new ErrorBuilder({ values });

  errorBuilder = validateEssentials({ errorBuilder, values });

  if (tila === JULKAISUTILA.TALLENNETTU) {
    return errorBuilder.getErrors();
  }

  errorBuilder = validateCommon({ values, errorBuilder });

  if (KORKEAKOULUKOULUTUSTYYPIT.includes(koulutustyyppi)) {
    errorBuilder = validateKorkeakoulu({ values, errorBuilder });
  }

  return errorBuilder.getErrors();
};
