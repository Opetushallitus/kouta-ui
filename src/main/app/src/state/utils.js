import { getFormValues, startSubmit, stopSubmit } from 'redux-form';

import { isNonEmptyObject } from '../utils';
import { createSavingErrorToast, createSavingSuccessToast } from './toaster';

const identity = arg => arg;

export const submitAction = ({
  formName,
  validate,
  createSaveAction,
  getValidationArguments = identity,
}) => async (dispatch, getState) => {
  const state = getState();
  const values = getFormValues(formName)(state);
  const errors = validate(getValidationArguments({ values }));

  dispatch(startSubmit(formName));

  if (!isNonEmptyObject(errors)) {
    dispatch(stopSubmit(formName, errors));
    dispatch(createSavingErrorToast());
    return;
  }

  const saveAction = await createSaveAction({ values });

  try {
    const data = await dispatch(saveAction);
    dispatch(stopSubmit(formName));
    dispatch(createSavingSuccessToast());

    return data;
  } catch (e) {
    dispatch(stopSubmit(formName));
    dispatch(createSavingErrorToast());

    throw e;
  }
};
