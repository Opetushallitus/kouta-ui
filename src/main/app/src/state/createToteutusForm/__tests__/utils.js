import { getToteutusByValues, getValuesByToteutus } from '../utils';

test('getToteutusByValues returns correct toteutus given form values', () => {
  const toteutus = getToteutusByValues({
    nimi: {
      name: {
        fi: 'Fi nimi',
        sv: 'Sv nimi',
      },
    },
    kieliversiot: {
      languages: ['fi', 'sv'],
    },
    jarjestamispaikat: {
      jarjestajat: ['org1', 'org2'],
    },
    jarjestamistiedot: {
      kuvaus: {
        fi: 'Fi kuvaus',
        sv: 'Sv kuvaus',
      },
      maksullisuus: {
        tyyppi: 'kylla',
        maksu: '50.50',
      },
      opetustapa: ['opetustapa_1#1', 'opetustapa_2#1'],
      opetusaika: ['opetusaika_1#1', 'opetusaika_2#1'],
      opetuskieli: ['opetuskieli_1#1', 'opetuskieli_2#1'],
      opetusaikaKuvaus: {
        fi: 'Fi aikakuvaus',
        sv: 'Sv aikakuvaus',
      },
      opetustapaKuvaus: {
        fi: 'Fi tapakuvaus',
        sv: 'Sv tapakuvaus',
      },
      opetuskieliKuvaus: {
        fi: 'Fi kielikuvaus',
        sv: 'Sv kielikuvaus',
      },
      maksullisuusKuvaus: {
        fi: 'Fi maksullisuuskuvaus',
        sv: 'Sv maksullisuuskuvaus',
      },
      osiot: [{ value: 'osio_1#1' }, { value: 'osio_2#1' }],
      osioKuvaukset: {
        'osio_1#1': {
          fi: 'Fi kuvaus1',
          sv: 'Sv kuvaus1',
        },
        'osio_2#1': {
          fi: 'Fi kuvaus2',
          sv: 'Sv kuvaus2',
        },
      },
      alkamiskausi: {
        kausi: 'kausi_1#1',
        vuosi: { value: '2020' },
      },
      alkamiskausiKuvaus: {
        fi: 'Fi kausikuvaus',
        sv: 'Sv kausikuvaus',
      },
      onkoStipendia: true,
      stipendinMaara: { fi: '20e', sv: '20kr' },
      stipendinKuvaus: {
        fi: 'Fi stipendikuvaus',
        sv: 'Sv stipendikuvaus',
      },
    },
    nayttamistiedot: {
      ammattinimikkeet: {
        fi: [{ value: 'nimike1' }],
        sv: [{ value: 'nimike2' }],
      },
      avainsanat: {
        fi: [{ value: 'avainsana1' }],
        sv: [{ value: 'avainsana2' }],
      },
    },
    yhteyshenkilot: [
      {
        nimi: { fi: 'Fi nimi', sv: 'Sv nimi' },
        titteli: { fi: 'Fi titteli', sv: 'Sv titteli' },
        sahkoposti: { fi: 'Fi sähköposti', sv: 'Sv sähköposti' },
        puhelinnumero: { fi: 'Fi puhelinnumero', sv: 'Sv puhelinnumero' },
        verkkosivu: { fi: 'Fi verkkosivu', sv: 'Sv verkkosivu' },
      },
    ],
    osaamisalat: {
      osaamisalat: ['osaamisala1', 'osaamisala2'],
      osaamisalaLinkit: {
        osaamisala1: {
          fi: 'Fi linkki1',
          sv: 'Sv linkki1',
        },
        osaamisala2: {
          fi: 'Fi linkki2',
          sv: 'Sv linkki2',
        },
      },
      osaamisalaLinkkiOtsikot: {
        osaamisala1: {
          fi: 'Fi otsikko1',
          sv: 'Sv otsikko1',
        },
        osaamisala2: {
          fi: 'Fi otsikko2',
          sv: 'Sv otsikko2',
        },
      },
    },
    ylemmanKorkeakoulututkinnonOsaamisalat: [
      {
        nimi: { fi: 'Fi ylempinimi', sv: 'Sv ylempinimi' },
        kuvaus: { fi: 'Fi ylempikuvaus', sv: 'Sv ylempikuvaus' },
        linkki: { fi: 'Fi ylempilinkki', sv: 'Sv ylempilinkki' },
        otsikko: { fi: 'Fi ylempiotsikko', sv: 'Sv ylempiotsikko' },
      },
    ],
    alemmanKorkeakoulututkinnonOsaamisalat: [
      {
        nimi: { fi: 'Fi alempinimi', sv: 'Sv alempinimi' },
        kuvaus: { fi: 'Fi alempikuvaus', sv: 'Sv alempikuvaus' },
        linkki: { fi: 'Fi alempilinkki', sv: 'Sv alempilinkki' },
        otsikko: { fi: 'Fi alempiotsikko', sv: 'Sv alempiotsikko' },
      },
    ],
    kuvaus: {
      kuvaus: {
        fi: 'Fi kuvaus',
        sv: 'Sv kuvaus',
      },
    },
  });

  expect(toteutus).toMatchSnapshot();
});

test('getValuesByToteutus returns correct form values given toteutus', () => {
  const values = getValuesByToteutus({
    kielivalinta: ['fi', 'sv'],
    metadata: {
      alemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: {
            fi: 'Fi alempikuvaus',
            sv: 'Sv alempikuvaus',
          },
          linkki: {
            fi: 'Fi alempilinkki',
            sv: 'Sv alempilinkki',
          },
          nimi: {
            fi: 'Fi alempinimi',
            sv: 'Sv alempinimi',
          },
          otsikko: {
            fi: 'Fi alempiotsikko',
            sv: 'Sv alempiotsikko',
          },
        },
      ],
      ammattinimikkeet: [
        {
          arvo: 'nimike1',
          kieli: 'fi',
        },
        {
          arvo: 'nimike2',
          kieli: 'sv',
        },
      ],
      asiasanat: [
        {
          arvo: 'avainsana1',
          kieli: 'fi',
        },
        {
          arvo: 'avainsana2',
          kieli: 'sv',
        },
      ],
      kuvaus: {
        fi: 'Fi kuvaus',
        sv: 'Sv kuvaus',
      },
      opetus: {
        alkamisaikaKuvaus: {
          fi: 'Fi kausikuvaus',
          sv: 'Sv kausikuvaus',
        },
        alkamiskausiKoodiUri: 'kausi_1#1',
        alkamisvuosi: '2020',
        lisatiedot: [
          {
            otsikkoKoodiUri: 'osio_1#1',
            teksti: {
              fi: 'Fi kuvaus1',
              sv: 'Sv kuvaus1',
            },
          },
          {
            otsikkoKoodiUri: 'osio_2#1',
            teksti: {
              fi: 'Fi kuvaus2',
              sv: 'Sv kuvaus2',
            },
          },
        ],
        lukuvuosimaksu: 40.5,
        lukuvuosimaksuKuvaus: {
          fi: 'Fi maksukuvaus',
          sv: 'Sv maksukuvaus',
        },
        maksullisuusKuvaus: {
          fi: 'Fi maksullisuuskuvaus',
          sv: 'Sv maksullisuuskuvaus',
        },
        maksunMaara: 50.25,
        onkoLukuvuosimaksua: true,
        onkoMaksullinen: true,
        onkoStipendia: true,
        opetusaikaKoodiUrit: ['opetusaika_1#1'],
        opetusaikaKuvaus: {
          fi: 'Fi aikakuvaus',
          sv: 'Sv aikakuvaus',
        },
        opetuskieletKuvaus: {
          fi: 'Fi kielikuvaus',
          sv: 'Sv kielikuvaus',
        },
        opetuskieliKoodiUrit: ['opetuskieli_1#1', 'opetuskieli_2#1'],
        opetustapaKoodiUrit: ['opetustapa_1#1', 'opetustapa_2#1'],
        opetustapaKuvaus: {
          fi: 'Fi tapakuvaus',
          sv: 'Sv tapakuvaus',
        },
        stipendinKuvaus: {
          fi: 'Fi stipendikuvaus',
          sv: 'Sv stipendikuvaus',
        },
        stipendinMaara: {
          fi: '20e',
          sv: '20kr',
        },
      },
      osaamisalat: [
        {
          koodi: 'osaamisala1',
          linkki: {
            fi: 'Fi linkki1',
            sv: 'Sv linkki1',
          },
          otsikko: {
            fi: 'Fi otsikko1',
            sv: 'Sv otsikko1',
          },
        },
        {
          koodi: 'osaamisala2',
          linkki: {
            fi: 'Fi linkki2',
            sv: 'Sv linkki2',
          },
          otsikko: {
            fi: 'Fi otsikko2',
            sv: 'Sv otsikko2',
          },
        },
      ],
      yhteyshenkilot: [
        {
          nimi: {
            fi: 'Fi nimi',
            sv: 'Sv nimi',
          },
          puhelinnumero: {
            fi: 'Fi puhelinnumero',
            sv: 'Sv puhelinnumero',
          },
          sahkoposti: {
            fi: 'Fi sähköposti',
            sv: 'Sv sähköposti',
          },
          titteli: {
            fi: 'Fi titteli',
            sv: 'Sv titteli',
          },
          wwwSivu: {
            fi: 'Fi verkkosivu',
            sv: 'Sv verkkosivu',
          },
        },
      ],
      ylemmanKorkeakoulututkinnonOsaamisalat: [
        {
          kuvaus: {
            fi: 'Fi ylempikuvaus',
            sv: 'Sv ylempikuvaus',
          },
          linkki: {
            fi: 'Fi ylempilinkki',
            sv: 'Sv ylempilinkki',
          },
          nimi: {
            fi: 'Fi ylempinimi',
            sv: 'Sv ylempinimi',
          },
          otsikko: {
            fi: 'Fi ylempiotsikko',
            sv: 'Sv ylempiotsikko',
          },
        },
      ],
    },
    nimi: {
      fi: 'Fi nimi',
      sv: 'Sv nimi',
    },
    tarjoajat: ['org1', 'org2'],
  });

  expect(values).toMatchSnapshot();
});
