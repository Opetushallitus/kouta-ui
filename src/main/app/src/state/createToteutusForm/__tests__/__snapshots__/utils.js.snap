// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getToteutusByValues returns correct toteutus given form values 1`] = `
Object {
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "metadata": Object {
    "alemmanKorkeakoulututkinnonOsaamisalat": Array [
      Object {
        "kuvaus": Object {
          "fi": "Fi alempikuvaus",
          "sv": "Sv alempikuvaus",
        },
        "linkki": Object {
          "fi": "Fi alempilinkki",
          "sv": "Sv alempilinkki",
        },
        "nimi": Object {
          "fi": "Fi alempinimi",
          "sv": "Sv alempinimi",
        },
        "otsikko": Object {
          "fi": "Fi alempiotsikko",
          "sv": "Sv alempiotsikko",
        },
      },
    ],
    "ammattinimikkeet": Array [
      Object {
        "arvo": "nimike1",
        "kieli": "fi",
      },
      Object {
        "arvo": "nimike2",
        "kieli": "sv",
      },
    ],
    "asiasanat": Array [
      Object {
        "arvo": "avainsana1",
        "kieli": "fi",
      },
      Object {
        "arvo": "avainsana2",
        "kieli": "sv",
      },
    ],
    "kuvaus": Object {
      "fi": "Fi kuvaus",
      "sv": "Sv kuvaus",
    },
    "opetus": Object {
      "alkamisaikaKuvaus": Object {
        "fi": "Fi kausikuvaus",
        "sv": "Sv kausikuvaus",
      },
      "alkamiskausiKoodiUri": "kausi_1#1",
      "alkamisvuosi": "2020",
      "lisatiedot": Array [
        Object {
          "otsikkoKoodiUri": "osio_1#1",
          "teksti": Object {
            "fi": "Fi kuvaus1",
            "sv": "Sv kuvaus1",
          },
        },
        Object {
          "otsikkoKoodiUri": "osio_2#1",
          "teksti": Object {
            "fi": "Fi kuvaus2",
            "sv": "Sv kuvaus2",
          },
        },
      ],
      "lukuvuosimaksu": null,
      "maksullisuusKuvaus": Object {
        "fi": "Fi maksullisuuskuvaus",
        "sv": "Sv maksullisuuskuvaus",
      },
      "maksunMaara": 50.5,
      "onkoLukuvuosimaksua": false,
      "onkoMaksullinen": true,
      "onkoStipendia": true,
      "opetusaikaKoodiUrit": Array [
        "opetusaika_1#1",
        "opetusaika_2#1",
      ],
      "opetusaikaKuvaus": Object {
        "fi": "Fi aikakuvaus",
        "sv": "Sv aikakuvaus",
      },
      "opetuskieletKuvaus": Object {
        "fi": "Fi kielikuvaus",
        "sv": "Sv kielikuvaus",
      },
      "opetuskieliKoodiUrit": Array [
        "opetuskieli_1#1",
        "opetuskieli_2#1",
      ],
      "opetustapaKoodiUrit": Array [
        "opetustapa_1#1",
        "opetustapa_2#1",
      ],
      "opetustapaKuvaus": Object {
        "fi": "Fi tapakuvaus",
        "sv": "Sv tapakuvaus",
      },
      "stipendinKuvaus": Object {
        "fi": "Fi stipendikuvaus",
        "sv": "Sv stipendikuvaus",
      },
      "stipendinMaara": Object {
        "fi": "20e",
        "sv": "20kr",
      },
    },
    "osaamisalat": Array [
      Object {
        "koodi": "osaamisala1",
        "linkki": Object {
          "fi": "Fi linkki1",
          "sv": "Sv linkki1",
        },
        "otsikko": Object {
          "fi": "Fi otsikko1",
          "sv": "Sv otsikko1",
        },
      },
      Object {
        "koodi": "osaamisala2",
        "linkki": Object {
          "fi": "Fi linkki2",
          "sv": "Sv linkki2",
        },
        "otsikko": Object {
          "fi": "Fi otsikko2",
          "sv": "Sv otsikko2",
        },
      },
    ],
    "yhteyshenkilot": Array [
      Object {
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "puhelinnumero": Object {
          "fi": "Fi puhelinnumero",
          "sv": "Sv puhelinnumero",
        },
        "sahkoposti": Object {
          "fi": "Fi sähköposti",
          "sv": "Sv sähköposti",
        },
        "titteli": Object {
          "fi": "Fi titteli",
          "sv": "Sv titteli",
        },
        "wwwSivu": Object {
          "fi": "Fi verkkosivu",
          "sv": "Sv verkkosivu",
        },
      },
    ],
    "ylemmanKorkeakoulututkinnonOsaamisalat": Array [
      Object {
        "kuvaus": Object {
          "fi": "Fi ylempikuvaus",
          "sv": "Sv ylempikuvaus",
        },
        "linkki": Object {
          "fi": "Fi ylempilinkki",
          "sv": "Sv ylempilinkki",
        },
        "nimi": Object {
          "fi": "Fi ylempinimi",
          "sv": "Sv ylempinimi",
        },
        "otsikko": Object {
          "fi": "Fi ylempiotsikko",
          "sv": "Sv ylempiotsikko",
        },
      },
    ],
  },
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "tarjoajat": Object {
    "jarjestajat": Array [
      "org1",
      "org2",
    ],
  },
}
`;

exports[`getValuesByToteutus returns correct form values given toteutus 1`] = `
Object {
  "alemmanKorkeakoulututkinnonOsaamisalat": Array [
    Object {
      "kuvaus": Object {
        "fi": "Fi alempikuvaus",
        "sv": "Sv alempikuvaus",
      },
      "linkki": Object {
        "fi": "Fi alempilinkki",
        "sv": "Sv alempilinkki",
      },
      "nimi": Object {
        "fi": "Fi alempinimi",
        "sv": "Sv alempinimi",
      },
      "otsikko": Object {
        "fi": "Fi alempiotsikko",
        "sv": "Sv alempiotsikko",
      },
    },
  ],
  "jarjestamispaikat": Array [
    "org1",
    "org2",
  ],
  "jarjestamistiedot": Object {
    "alkamiskausi": Object {
      "kausi": "kausi_1#1",
      "vuosi": Object {
        "value": "2020",
      },
    },
    "alkamiskausiKuvaus": Object {
      "fi": "Fi kausikuvaus",
      "sv": "Sv kausikuvaus",
    },
    "kuvaus": Object {
      "fi": "Fi kuvaus",
      "sv": "Sv kuvaus",
    },
    "lukuvuosimaksu": 40.5,
    "lukuvuosimaksuKuvaus": Object {
      "fi": "Fi maksukuvaus",
      "sv": "Sv maksukuvaus",
    },
    "maksullisuus": Object {
      "maksu": "40.5",
      "tyyppi": "lukuvuosimaksu",
    },
    "maksullisuusKuvaus": Object {
      "fi": "Fi maksullisuuskuvaus",
      "sv": "Sv maksullisuuskuvaus",
    },
    "maksumaara": 50.25,
    "onkoLukuvuosimaksua": true,
    "onkoStipendia": true,
    "opetusaika": Array [
      "opetusaika_1#1",
    ],
    "opetusaikaKuvaus": Object {
      "fi": "Fi aikakuvaus",
      "sv": "Sv aikakuvaus",
    },
    "opetuskieli": Array [
      "opetuskieli_1#1",
      "opetuskieli_2#1",
    ],
    "opetuskieliKuvaus": Object {
      "fi": "Fi kielikuvaus",
      "sv": "Sv kielikuvaus",
    },
    "opetustapa": Array [
      "opetustapa_1#1",
      "opetustapa_2#1",
    ],
    "opetustapaKuvaus": Object {
      "fi": "Fi tapakuvaus",
      "sv": "Sv tapakuvaus",
    },
    "osioKuvaukset": Object {
      "osio_1#1": Object {
        "fi": "Fi kuvaus1",
        "sv": "Sv kuvaus1",
      },
      "osio_2#1": Object {
        "fi": "Fi kuvaus2",
        "sv": "Sv kuvaus2",
      },
    },
    "osiot": Array [
      Object {
        "value": "osio_1#1",
      },
      Object {
        "value": "osio_2#1",
      },
    ],
    "stipendinKuvaus": Object {
      "fi": "Fi stipendikuvaus",
      "sv": "Sv stipendikuvaus",
    },
    "stipendinMaara": Object {
      "fi": "20e",
      "sv": "20kr",
    },
  },
  "kieliversiot": Object {
    "languages": Array [
      "fi",
      "sv",
    ],
  },
  "kuvaus": Object {
    "kuvaus": Object {
      "fi": "Fi kuvaus",
      "sv": "Sv kuvaus",
    },
  },
  "nayttamistiedot": Object {
    "ammattinimikkeet": Object {
      "fi": Array [
        Object {
          "label": "nimike1",
          "value": "nimike1",
        },
      ],
      "sv": Array [
        Object {
          "label": "nimike2",
          "value": "nimike2",
        },
      ],
    },
    "avainsanat": Object {
      "fi": Array [
        Object {
          "label": "avainsana1",
          "value": "avainsana1",
        },
      ],
      "sv": Array [
        Object {
          "label": "avainsana2",
          "value": "avainsana2",
        },
      ],
    },
  },
  "nimi": Object {
    "name": Object {
      "fi": "Fi nimi",
      "sv": "Sv nimi",
    },
  },
  "osaamisalat": Object {
    "osaamisalaLinkit": Object {
      "osaamisala1": Object {
        "fi": "Fi linkki1",
        "sv": "Sv linkki1",
      },
      "osaamisala2": Object {
        "fi": "Fi linkki2",
        "sv": "Sv linkki2",
      },
    },
    "osaamisalaLinkkiOtsikot": Object {
      "osaamisala1": Object {
        "fi": "Fi otsikko1",
        "sv": "Sv otsikko1",
      },
      "osaamisala2": Object {
        "fi": "Fi otsikko2",
        "sv": "Sv otsikko2",
      },
    },
    "osaamisalat": Array [
      "osaamisala1",
      "osaamisala2",
    ],
  },
  "yhteyshenkilot": Array [
    Object {
      "nimi": Object {
        "fi": "Fi nimi",
        "sv": "Sv nimi",
      },
      "puhelinnumero": Object {
        "fi": "Fi puhelinnumero",
        "sv": "Sv puhelinnumero",
      },
      "sahkoposti": Object {
        "fi": "Fi sähköposti",
        "sv": "Sv sähköposti",
      },
      "titteli": Object {
        "fi": "Fi titteli",
        "sv": "Sv titteli",
      },
      "verkkosivu": Object {
        "fi": "Fi verkkosivu",
        "sv": "Sv verkkosivu",
      },
    },
  ],
  "ylemmanKorkeakoulututkinnonOsaamisalat": Array [
    Object {
      "kuvaus": Object {
        "fi": "Fi ylempikuvaus",
        "sv": "Sv ylempikuvaus",
      },
      "linkki": Object {
        "fi": "Fi ylempilinkki",
        "sv": "Sv ylempilinkki",
      },
      "nimi": Object {
        "fi": "Fi ylempinimi",
        "sv": "Sv ylempinimi",
      },
      "otsikko": Object {
        "fi": "Fi ylempiotsikko",
        "sv": "Sv ylempiotsikko",
      },
    },
  ],
}
`;
